x/marker/keeper/msg_server.go

```diff
@@ -734,3 +737,30 @@ func (k msgServer) UpdateForcedTransfer(goCtx context.Context, msg *types.MsgUpd
 
 	return &types.MsgUpdateForcedTransferResponse{}, nil
 }
+
+// SetAccountData sets the accountdata for a denom. Signer must have deposit authority.
+func (k msgServer) SetAccountData(goCtx context.Context, msg *types.MsgSetAccountDataRequest) (*types.MsgSetAccountDataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	marker, err := k.GetMarkerByDenom(ctx, msg.Denom)
+	if err != nil {
+		return nil, fmt.Errorf("could not get %s marker: %w", msg.Denom, err)
+	}
+
+	if msg.Signer == k.GetAuthority() {
+		if !marker.HasGovernanceEnabled() {
+			return nil, fmt.Errorf("%s marker does not allow governance control", msg.Denom)
+		}
+	} else {
+		if !marker.HasAccess(msg.Signer, types.Access_Deposit) {
```

Comment:
I didn't use `admin` because that's supposedly only for the ability to manage permissions on a marker. It's not exactly named the best thing though primarily due to confusion like this.

The `deposit` permission is what's used in the metadata module when setting a marker as the scope's value owner. Someone signing the message for that update has to have `deposit` permission on that marker. Since that's kind of like associating data with a marker, it felt like the best fit for a marker's account data too.

It would also make sense if `withdraw` permission were needed to set a marker's account data to `""`, but I opted to keep it simple. That might be worth revisiting later, though, depending on users' needs.

Reply:
In this context "deposit" is used for that permission flow because we are "depositing" the NFT into the marker as an asset.  Admin == Grant originally... (but was renamed away from that because grant-grants-grants was exceptionally confusing in naming).

 (╯°□°）╯︵ ┻━┻